### ISSUE TITLE
Feature request: flow-based activation of alarm

### ISSUE THREAD

**Issue by rkokkelk** [daneedk/com.uc.heimdall#19]
Hi,

First of all great app. I've manually tried to implement the same functionality with a lot of different flows, but your app makes it a lot easier.

The only feature that I'm currently missing is the ability to change whether an alarm is activated when a motion sensor is changed based on some conditions. For example, I've got a Roomba and it usually works when I'm not home and when my alarm is activated. But off course when the Roomba is working I don't need the alarm to go off.

So what I was thinking, is implementing the functionality to allow a flow which is triggered when a sensor has been activated, and which has to successfully run before the alarm actually goes off. Then it is possible to implement some boolean expressions within the flow in order to prevent the alarm to go off when the Roomba is working. 

Let me know what you think.

**Comment by daneedk** [comment]
For this particular situation I would make this flow:

![image](https://user-images.githubusercontent.com/16610696/39990778-c6b0e724-576d-11e8-99c6-339386a584f0.png)

You can make the action taken when the **_The alarm is activated_** card is triggered conditional in a flow so I don't think extra functionality is needed in the Heimdall app.

**Comment by rkokkelk** [comment]
Ah that is a great tip, and will definitely work. However, the Heimdall will register that the alarm has gone off and will have logged it, etc. So I'm looking for a way that the alarm will simple not be activated depending on certain conditions. 

Is something you might be interested in working on, or in accepting it as functionality. When I have some spare time I could make a PR which adds this specific functionality.

**Comment by rkokkelk** [comment]
Hi,

I was actually hoping whether your are interested in this functionality. So that if I make it and provide it in a PR it will get merged. I've used the above described functionality and I found some downsides. 

Mainly, when I use the Roomba which starts in the morning, the alarm is triggered. I do not receive a notification, but due to the fact that the alarm is already triggered if someone actually breaks in, even when the Roomba is stopped, nothing happens because the alarm is already triggered. So I'm still looking for the ideal solution in which a flow prevent the alarm being triggered at all.
